meson.build

```diff
@@ -0,0 +1,1449 @@
+# libc, per section 2.3 of release info at https://sourceware.org/glibc/wiki/Release/2.34.  This means, if we're using
+# the GNU tool chain and libc is v2.34 or newer, then we should NOT look for libdl, as we won't find it!  To find the
+# version of libc, you execute 'ldd --version', which gives multi-line output, of which the first line will be something
+# such as:
+#    ldd (Ubuntu GLIBC 2.35-0ubuntu3.1) 2.35
+# In this case, that means libc is version 2.35
+#
+# If we _don't_ need libdl, we just create an "unrequired" dependency.  This saves having to repeat a bunch of logic
+# later on when we get to the main build.
+#
+needLibdl = false
+if compiler.get_id() == 'gcc'
+   lddOutput = run_command('ldd', '--version', check: true).stdout().strip()
+   lddOutputLine1 = lddOutput.split('\n')[0]
+   message('ldd --version gives:', lddOutputLine1)
+   libcVersion = lddOutputLine1.split(' ')[-1]
+   message('libc version is:', libcVersion)
+   if libcVersion.version_compare('<2.34')
+      needLibdl = true
+   endif
+endif
+dlDependency = dependency('dl', required : needLibdl, static : true)
+#
+# On newer versions of Meson (0.62.0 and newer), the dependency has special-case code for dealing with libdl being built
+# into the compiler.  This means on Mac, for instance, we can have dlDependency.found() is true without there being any
+# cmake variable such as PACKAGE_LIBRARIES (because the library was found by Meson special case code, not by cmake).
+#
+# This is why we have to supply a default dummy value, which we then check for.
+#
+if dlDependency.found()
+   dlLibraryPath = dlDependency.get_variable(default_value : '', cmake : 'PACKAGE_LIBRARIES')
+   message('dlLibraryPath is:', dlLibraryPath)
+   if dlLibraryPath != ''
+      sharedLibraryPaths += dlLibraryPath
+   endif
+endif
+
+# Note that, unlike, say, the parameters to include_directories(), the dirs argument to find_library() must be absolute
+# paths
+libbacktraceDir = join_paths(meson.project_source_root(), 'third-party/libbacktrace/.libs')
+backtraceDependency = compiler.find_library('backtrace',
+                                            required : true,
+                                            static : true,
+                                            dirs : [libbacktraceDir])
+
+#======================================== Find the other libraries we depend on ========================================
+#
+# See https://mesonbuild.com/Reference-manual_returned_dep.html for what info we can pull from a dependency object
+#
+# For BeerXML processing we need Xerces-C++ and Xalan-C++.  Meson can find both of these automatically using CMake's
+# find_package(), as long as (a) CMake is installed(!) and (b) we provide the right library names ('XercesC' per
+# https://cmake.org/cmake/help/latest/module/FindXercesC.html and 'XalanC' per
+# https://cmake.org/cmake/help/latest/module/FindXalanC.html)
```

Comment:
Probably the official way to search for both is with `dependency('xerces-c')` and `dependency('xalan-c')`. xalan-c provides a cmake package, but xerces-c doesn't.

Both provide pkg-config files which are lowercased and have a hyphen, so you need to use the meson 0.60.0 support for `dependency('xerces-c', 'XercesC')` with multiple possible names for the same dependency call, if you want to be able to find either one. If you can find it with pkg-config, you don't need cmake installed. You can get the 'libdir' variable from the dependency if its `.type_name() == 'pkgconfig'`.

Reply:
Ah, OK, that's interesting.
AFAICT, the downside of using `dependency('xerces-c', 'XercesC')` is that, if the dependency is found natively by Meson (rather than calling out to CMake's `find_package()`), then we don't have a way to know what directory the library was found in, which is rather nice to have for passing out to the packaging scripts.  But very happy to be corrected about this!